internal/channelz/funcs.go

```diff
@@ -295,6 +295,21 @@ type TraceEventDesc struct {
 
 // AddTraceEvent adds trace related to the entity with specified id, using the provided TraceEventDesc.
 func AddTraceEvent(id int64, desc *TraceEventDesc) {
+	for _, d := range [2]*TraceEventDesc{desc, desc.Parent} {
+		if d == nil {
```

Comment:
Can these be nested deeper than two?  Maybe one day?  Maybe we should do this instead:

```go
for d := desc; d != nil; d = d.Parent {
  switch desc.Severity {
  ...
  }
}
```

Reply:
Fixed this.